src/main/java/com/autotune/analyzer/plots/PlotManager.java

```diff
@@ -58,37 +60,83 @@ public PlotData.PlotsData generatePlots() {
             calendar.add(Calendar.MILLISECOND, (int) millisecondsToAdd);
             // Convert the modified Calendar back to a Timestamp
             Timestamp newTimestamp = new Timestamp(calendar.getTimeInMillis());
-            PlotData.UsageData cpuUsage = getUsageData(sortedResultsHashMap.subMap(newTimestamp, true, incrementStartTime, true), AnalyzerConstants.MetricName.cpuUsage, "cores");
-            PlotData.UsageData memoryUsage = getUsageData(sortedResultsHashMap.subMap(newTimestamp, true, incrementStartTime, true), AnalyzerConstants.MetricName.memoryUsage, "MiB");
+            PlotData.UsageData cpuUsage = getUsageData(sortedResultsHashMap.subMap(newTimestamp, true,
+                    incrementStartTime,false), AnalyzerConstants.MetricName.cpuUsage);
+            PlotData.UsageData memoryUsage = getUsageData(sortedResultsHashMap.subMap(newTimestamp, true,
+                    incrementStartTime, false), AnalyzerConstants.MetricName.memoryUsage);
             plotsDataMap.put(newTimestamp, new PlotData.PlotPoint(cpuUsage, memoryUsage));
             incrementStartTime = newTimestamp;
         }
 
         return new PlotData.PlotsData(recommendationTerm.getPlots_datapoints(), plotsDataMap);
     }
 
-    PlotData.UsageData getUsageData(Map<Timestamp, IntervalResults> resultInRange, AnalyzerConstants.MetricName metricName, String format) {
-        // Extract CPU values
-        List<Double> cpuValues = resultInRange.values().stream()
+    PlotData.UsageData getUsageData(Map<Timestamp, IntervalResults> resultInRange, AnalyzerConstants.MetricName metricName) {
+        // stream through the results value and extract the format and the metric values
+        PlotData.UsageData usageData = resultInRange.values().stream()
                 .filter(intervalResults -> intervalResults.getMetricResultsMap().containsKey(metricName))
-                .mapToDouble(intervalResults -> {
+                .map(intervalResults -> {
                     MetricResults metricResults = intervalResults.getMetricResultsMap().get(metricName);
-                    return (metricResults != null && metricResults.getAggregationInfoResult() != null) ? metricResults.getAggregationInfoResult().getSum() : 0.0;
+                    double metricUsageAvg = Optional.ofNullable(metricResults)
+                            .map(MetricResults::getAggregationInfoResult)
+                            .map(MetricAggregationInfoResults::getAvg)
+                            .orElse(0.0);
+                    double metricUsageMax = Optional.ofNullable(metricResults)
+                            .map(MetricResults::getAggregationInfoResult)
+                            .map(MetricAggregationInfoResults::getMax)
+                            .orElse(0.0);
+                    double metricUsageSum = Optional.ofNullable(metricResults)
```

Comment:
Plot data needs to be calculated based on the  derived cpuUsage and memUsage metrics which we calculate in recommendations logic.

Reply:
does it mean 
cpu_usage = Sum_of_CPU_Usage + Sum_of_CPU_Throttle ?

Reply:
@khansaad intention behind plots is to illustrate to users the CPU consumption. How does this correlate with the recommendation logic?